src/PubSub/Connection/Rest.php

```diff
@@ -38,10 +41,20 @@ class Rest implements ConnectionInterface
      */
     public function __construct(array $config = [])
     {
+        $baseUri = self::BASE_URI;
+
+        $pubSubEmulatorHost = getenv('PUBSUB_EMULATOR_HOST');
+        if ($pubSubEmulatorHost && $uriComponents = parse_url($pubSubEmulatorHost)) {
```

Comment:
Can you please move the assignment out of the conditional and onto its own line inside the block?